docs/calendar-review.md

```diff
@@ -0,0 +1,212 @@
+date.with({ day: 1 }); // => 1 Adar I 5779
+date.year; // => 5779
+date.monthCode; // => M05L
+date.month; // => 6
+date.day; // => 2
+date.inLeapYear; // => true
+date.calendar.id; // => hebrew
+inTwoMonths = date.add({ months: 4 }); // => 23 Sivan 5779
+inTwoMonths.withCalendar('iso8601'); // => 2019-06-26
+date.until(inTwoMonths, { largestUnit: 'months' }); // => P4M
+```
+
+By definition, creating a bag of fields with year, month, and day requires choosing a calendar system. We elide the ISO calendar if no other calendar system is specified in this situation, and it is the only place Temporal performs calendar elision:
+
+```javascript
+Temporal.PlainDate.from({ year: 2019, month: 2, day: 28 });
+```
+
+The following Temporal types all carry a calendar: `Temporal.ZonedDateTime`, `Temporal.PlainDateTime`, `Temporal.PlainDate`, `Temporal.PlainYearMonth`, `Temporal.PlainMonthDay`
+
+## The Unexpected Calendar Problem
+
+Integrating calendar awareness into many Temporal operations is essential for a global audience, but it introduces a new problem: if code is written to assume one calendar system, bugs may result if a different calendar system is introduced instead. Introducing unexpected calendars could also be a vector for malicious activity.
+
+This section describes this problem, which we call the “Unexpected Calendar Problem,” and how to mitigate it in userland Temporal code.
+
+### Problem Definition
+
+Below is a canonical example of this problem: a function which tests whether a particular date is the last day of the year. It assumes a calendar with 12 months and whose last month is 31 days long.
+
+```javascript
+function isLastDayOfYear(date) {
+  // Note: This is a BAD example!
+  return date.month === 12 && date.day === 31;
+}
+
+// works for ISO calendar
+isLastDayOfYear(Temporal.PlainDate.from('2019-12-31')); // => true
+isLastDayOfYear(Temporal.PlainDate.from('2020-01-01')); // => false
+
+// fails for some non-ISO calendars
+hebrewNewYearsEve = Temporal.PlainDate.from(
+  { year: 5780, monthCode: 'M12', day: 29, calendar: 'hebrew' });
+isLastDayOfYear(hebrewNewYearsEve); // => expected: true, actual: false
+```
+
+### Mitigating the Unexpected Calendar Problem
+
+There are two ways to avoid the Unexpected Calendar Problem: the preferred solution of writing calendar-safe code, or a fallback option to validate or coerce inputs to a known calendar.
+
+#### Writing Calendar-Safe Code
+
+The best way to mitigate the Unexpected Calendar Problem is to write “calendar-safe” code that works for all built-in calendars. For example, the function above can easily be written to be calendar-safe by using the `monthsInYear` and `daysInMonth` properties instead of hard-coding ISO constants. Developers who follow documented [best practices for writing calendar-safe Temporal code](https://tc39.es/proposal-temporal/docs/calendar.html#writing-cross-calendar-code) will be able to use the same code for all built-in calendars.
```

Comment:
```suggestion
The best way to mitigate the Unexpected Calendar Problem is to write "calendar-safe" code that works for all built-in calendars. For example, the function above can easily be written to be calendar-safe by using the `monthsInYear` and `daysInMonth` properties instead of hard-coding ISO constants. Developers who follow documented [best practices for writing calendar-safe Temporal code](https://tc39.es/proposal-temporal/docs/calendar.html#writing-cross-calendar-code) will be able to use the same code for all built-in calendars.
```